analyses/gistic-vs-focal-cn-file-preparation/01-GISTIC-tidy-data-prep.Rmd

```diff
@@ -0,0 +1,261 @@
+      fill = TRUE)
+  
+  # Tidy the data.frame into a key and value format where the detection peak
+  # region is the key and the genes in that region are the values
+  genes_df <- genes_df %>%
+    dplyr::rename(peak_region = X1) %>%
+    tidyr::gather("region", "gene_symbol", -peak_region) %>%
+    dplyr::select(-region) %>%
+    dplyr::filter(!(is.na(gene_symbol)))
+  
+}
+
+prepare_gene_level_gistic <- function(all_lesions_file,
+                                      amp_genes_file,
+                                      del_genes_file,
+                                      output_filename) {
+  
+  # Given the file paths to GISTIC's `all_lesion.cong_90.txt`,
+  # `amp_genes.conf_90.txt`, and `del_genes.conf_90.txt` files,
+  # read in and tidy this data into a data.frame that contains
+  # the amplified/deleted genes, the detection peak, and
+  # the IDs of the samples in the corresponding peaks along with
+  # with their CN status call.
+  #
+  # Args:
+  #   all_lesions_file: file path to GISTIC's `all_lesion.conf_90.txt` file
+  #   amp_genes_file: file path to GISTIC's `amp_genes.conf_90.txt` file
+  #   del_genes_file: file path to GISTIC's `del_genes.conf_90.txt` file
+  #   output_filename: string to save the output file as
+  #
+  # Return:
+  #   final_df: data.frame with the relevant data from the `all_lesions`,
+  #             `amp_genes`, and `del_genes` files
+  
+  # Read in files
+  gistic_all_lesions_df <- data.table::fread(all_lesions_file,
+                                             data.table = FALSE)
+  
+  gistic_amp_genes_df <- data.table::fread(amp_genes_file,
+                                           data.table = FALSE)
+  
+  gistic_del_genes_df <- data.table::fread(del_genes_file,
+                                           data.table = FALSE)
+  
+  # Run `get_genes_df` function on `amp_genes` and `del_genes` files to get
+  # a data.frame with just the genes and their correspondind detection peak
+  amp_genes_df <- get_genes_df(gistic_amp_genes_df)
+  
+  del_genes_df <- get_genes_df(gistic_del_genes_df)
+  
+  # Bind the rows from the above data.frames into one data.frame
+  final_df <- rbind(amp_genes_df, del_genes_df)
+  
+  
+  # Wrangle the GISTIC all lesions data to be in a comparable format with our CN calls
+  gistic_all_lesions_df <- gistic_all_lesions_df %>%
+    dplyr::select(
+      -c(
+        "Descriptor",
+        "Peak Limits",
+        "Region Limits",
+        "q values",
+        "Residual q values after removing segments shared with higher peaks",
+        "Broad or Focal",
+        "Amplitude Threshold"
+      )
+    ) %>%
+    as.data.frame() %>%
+    tidyr::gather("Kids_First_Biospecimen_ID",
+                  "status",-`Unique Name`,-`Wide Peak Limits`) %>%
+    arrange(desc(`Wide Peak Limits`)) %>%
+    dplyr::mutate(status = dplyr::case_when(
+      status %in% c("-1", "-1.2929") ~ "loss",
+      status %in% c("1", " 1", " 2", "2", " 1.0000") ~ "gain",
+      TRUE ~ "neutral"
```

Comment:
What this says to me is that something that should be a `double` is a `character`. For this to be robust to changes in the underlying values (e.g., what if some value becomes `-1.4597` another time we run it because the histology file changes), the logic should be 
```suggestion
      status < 0 ~ "loss",
      status > 0 ~ "gain",
      status == 0 ~ "neutral"
```

Whatever is causing the status values to be a character (and the inconsistencies in spacing) should be fixed before you get to this step.